src/mon/AuthMonitor.cc

```diff
@@ -1857,14 +1853,13 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
       goto done;
     }
 
-    if (!valid_caps(caps_vec, &ss)) {
+    if (!valid_caps(ceph_caps, &ss)) {
       err = -EINVAL;
       goto done;
     }
 
     map<string,bufferlist> newcaps;
-    for (vector<string>::iterator it = caps_vec.begin();
-	 it != caps_vec.end(); it += 2)
+    for (const auto& kv : ceph_caps)
       encode(*(it+1), newcaps[*it]);
```

Comment:
nit (as I think this gets reworked by further commits; otherwise wouldn't have compiled).

```suggestion
      encode(kv.second, newcaps[kv.first]);
```

Reply:
Yes.

Reply:
Fixed - https://github.com/ceph/ceph/pull/52008/commits/96b0015e557f4be7033e0bad9e60f07f25778434#diff-53eecdfcf56b08f0ec4b07dfb7eb394518f408e862c4a0173e7d40ebefb204e4R1863